### ISSUE TITLE
Expose docker.from_env() parameters

### ISSUE THREAD

**Issue by dpp23** [testcontainers/testcontainers-python#180]
`testcontainers.core.DockerClient` calls `docker.from_env()` it it's constructor. This method has a few parameters that will be useful to be exposed (see [reference](https://docker-py.readthedocs.io/en/stable/client.html?highlight=from_env#docker.client.from_env))

Ideally `testcontainers.core.DockerClient` and conversely `testcontainers.core.DockerContainer` should expose these parameters.

**Comment by timbmg** [comment]
Do I understand correctly, you want to be able to configure the docker client yourself? If so, you can do that already by changing the environment variables according to the reference you posted. For example you can set `DOCKER_HOST` to whatever URL you require.

**Comment by dpp23** [comment]
@timbmg no, this is not what I meant. I would like to be able to set `timeout` or `max_pool_size` as parameters passed to `docker.from_env`. I don't think these are read from env vars.

**Comment by timbmg** [comment]
I understand. I'd be happy to do a PR on this. 

So basically, we can add `**kwargs` to the `DockerClient` constructor and pass that to the `.from_env` method. Similarly, in the `DockerContainer` class,  we can pass the `kwargs` to the `DockerClient`.

What do you think @tillahoffmann @SergeyPirogov @KerstenBreuer @thedrow ?

**Comment by thedrow** [comment]
I think it could be useful to implement it this way but it doesn't really abstract the Docker client which isn't bad per-se if the project's scope is limited only to Docker and not other kind of containers (BSD Jails could be useful for example). However, we do need to consider that we might switch from the official docker client at some point since it may be phased out or since Docker Inc will unexpectedly go bankrupt etc.

I admit these risks are rather minor, but violating encapsulation needs better reasons than useful.

The design we're looking for is the ability to pass a ContainerClientConfiguration object which we can use for testing by mocking the object, provide configuration to multiple container implementations and maybe something I haven't thought of.

**Comment by tillahoffmann** [comment]
> So basically, we can add `**kwargs` to the `DockerClient` constructor and pass that to the `.from_env` method. Similarly, in the `DockerContainer` class, we can pass the `kwargs` to the `DockerClient`.

That sounds reasonable. But let's pass a dictionary `docker_client_kw` to `DockerContainer` and then unpack the dictionary in the call to `DockerClient` akin to `gridspec_kw` `matplotlib.pyplot.subplots`.
